application-glossary-deepl/src/main/resources/Glossary/Code/DeepL/GlossaryDeepLService.xml

```diff
@@ -0,0 +1,317 @@
+            return "${wikiPrefix}"
+        } else {
+            return "${glossariesPrefix}-${wikiPrefix}"
+        }
+    }
+    // Method used by different pages
+    def SynchronizeGlossaries()
+    {
+        getTranslator()
+        def context = xwikiContextProvider.get()
+
+        // Start by computing the list of glossaries that we can support
+        // Get the list of languages that are supported in the XWiki preferences
+        def reference = new DocumentReference(context.getWikiId(), "XWiki", "XWikiPreferences")
+        def preferencesDoc = context.getWiki().getDocument(reference, context)
+        def preferencesObject = preferencesDoc.getXObject(new ObjectReference('XWiki.XWikiPreferences', reference))
+        def missingLanguagesRaw = preferencesObject.getStringValue('languages').split(',')
+
+        def glossaryNamePrefix = GetGlossaryNamePrefix()
+
+        try {
+
+            def deeplGlossaries = translator.listGlossaries()
+            logger.debug('Fetched the list of registered glossaries : [{}]', deeplGlossaries)
+            def deeplSupportedGlossaries = translator.getGlossaryLanguages()
+            logger
+                .debug('Fetched the list of supported glossary language combinations : [{}]', deeplSupportedGlossaries)
+
+            Map&lt;String, List&lt;String&gt;&gt; languageCombinations = [:]
+            missingLanguagesRaw.each { sourceLanguage -&gt;
+                languageCombinations[sourceLanguage] = []
+                missingLanguagesRaw.each { targetLanguage -&gt;
+                    if (supportsLanguageCombination(deeplSupportedGlossaries, sourceLanguage, targetLanguage)) {
+                        languageCombinations[sourceLanguage].add(targetLanguage)
+                    }
+                }
+            }
+
+            languageCombinations.each { sourceLanguage, availableTargetLanguages -&gt;
+                availableTargetLanguages.each { targetLanguage -&gt;
+                    // Compute the name of the glossary
+                    def glossaryName = "${glossaryNamePrefix}-${getDeepLLanguage(sourceLanguage)}-${getDeepLLanguage(targetLanguage)}"
+
+                    logger.info('Synchronizing glossary [{}]', glossaryName)
+
+                    // Create the list of terms to be added to the glossary
+                    List&lt;String[]&gt; glossaryRawEntries =
+                        queryManager.createQuery(
+                            "select sourceDoc.title, targetDoc.title from XWikiDocument sourceDoc, XWikiDocument targetDoc, BaseObject glossaryObj where glossaryObj.name = sourceDoc.fullName and glossaryObj.name = targetDoc.fullName and glossaryObj.className = 'Glossary.Code.GlossaryClass' and ((sourceDoc.defaultLanguage = :sourceLanguage and sourceDoc.language = '') or sourceDoc.language = :sourceLanguage) and ((targetDoc.defaultLanguage = :targetLanguage and targetDoc.language = '') or targetDoc.language = :targetLanguage)",
+                            Query.HQL)
```

Comment:
What about permissions? Should any user be able to create a glossary entry that will be used for translations on the whole wiki?

Reply:
Should be fixed by 6d27ec5